src/lib/proof_system.ts

```diff
@@ -273,7 +275,7 @@ function ZkProgram<
   let publicInputType: ProvablePure<any> = config.publicInput! ?? Undefined;
   let publicOutputType: ProvablePure<any> = config.publicOutput! ?? Void;
 
-  let selfTag = { name: `Program${i++}` };
+  let selfTag = { name: config.name };
```

Comment:
Do we want to sanitize this input in any way since this will be written to disk? Some thoughts that come to mind:
1. Safe encoding (making sure only alphanumeric characters and separators are specified)
2. String length

These are just some thoughts, it's entirely up to you if you find them appropriate :D

Reply:
Yes very appropriate! I'll figure it out in the caching PR, maybe serialize them before using as filename but leave the namse string on the `ZkProgram` in original form